specification/eventgrid/data-plane/Microsoft.Storage/stable/2018-01-01/Storage.json

```diff
@@ -86,11 +95,167 @@
           "description": "An opaque string value representing the logical sequence of events for any particular blob name. Users can use standard string comparison to understand the relative sequence of two events on the same blob name.",
           "type": "string"
         },
+        "identity": {
+          "description": "The identity of the requester that triggered this event.",
+          "type": "string"
+        },
+        "storageDiagnostics": {
+          "description": "For service use only. Diagnostic data occasionally included by the Azure Storage service. This property should be ignored by event consumers.",
+          "type": "object"
+        }
+      }
+    },
+    "StorageDirectoryCreatedEventData": {
+      "description": "Schema of the Data property of an EventGridEvent for an Microsoft.Storage.DirectoryCreated event.",
+      "type": "object",
+      "properties": {
+        "api": {
+          "description": "The name of the API/operation that triggered this event.",
+          "type": "string"
+        },
+        "clientRequestId": {
+          "description": "A request id provided by the client of the storage API operation that triggered this event.",
+          "type": "string"
+        },
+        "requestId": {
+          "description": "The request id generated by the storage service for the storage API operation that triggered this event.",
+          "type": "string"
+        },
+        "eTag": {
+          "description": "The etag of the directory at the time this event was triggered.",
```

Comment:
readonly? If some properties are read_only, please add `read_only: true`

Reply:
Will leave this field as it is to keep it consistent with other schemas

Reply:
HI wanha-ms, what does "consistent with other schemas" mean? read only property need to have read_only: true to make sure downstream sdk generation correctly. It will cause customer failure if swagger not consistent with service behavior.

Reply:
This schema is used for EventGrid notifications for changes on Azure Storage. Whenever a file, directory were updated in a storage account, a notification is sent to EventGrid. When I sync with @MyronFanQiu earlier, my thought was, since it is a notification, in this case, all field in this schema is immutable, and should be readonly. However, I didn't see any field is specified in this way in earlier version, or other similar schema definitions. In this way, I think it is better to leave readonly as a future update, instead of in this PR.